atomate/vasp/firetasks/write_inputs.py

```diff
@@ -208,6 +209,34 @@ def run_task(self, fw_spec):
         potcar.write_file(self.get("output_filename", "POTCAR"))
 
 
+@explicit_serialize
+class WriteScanRelaxFromPrev(FiretaskBase):
+    """
+    Writes input files for a SCAN relaxation by constructing a new input set.
+    The purpose of this Firetask is to allow the KSPACING and smearing parameters
+    to be recalculated based on the bandgap from the PBE relaxation in the
+    SCAN relaxation workflow. Assumes that output files from a previous
+    (e.g., optimization) run can be accessed in current dir or prev_calc_dir. 
+
+    Required params:
+        (none)
+
+    Optional params (dict):
+        other_params: Dict of any keyword arguments supported by MPScanRelaxSet.
+    """
+    optional_params = ["other_params"]
+
+    def run_task(self, fw_spec):
+
+        kwargs = self.get("other_params")
+
+        os.chdir(os.getcwd())
+        vrun = Vasprun("vasprun.xml")
+        bandgap = vrun.get_band_structure().get_band_gap()["energy"]
+        structure = vrun.final_structure
+        vis = MPScanRelaxSet(structure, bandgap=bandgap, **kwargs)
```

Comment:
@utf I modeled `MPScanRelaxSet` on `MPRelaxSet`, which does not implement a `from_prev_calc` method. How is the information passed in that case? Is there a way I can achieve the desired behavior purely by modifying the FromPrev method in atomate without modifying the inputset?

Reply:
Good point! It seems like the MPRelaxSet doesn't support inheriting settings from a parent run either.

In that case, it seems unfair to force you to implement this also (although I think it would still be useful in some cases). 

I think this is something we should keep in mind for atomate v2!

Reply:
For your question. You could implement the FromPrev behavior in atomate but I would definitely advise against it. There should be a separation of input settings (controlled in pymatgen) and writing input sets for calculations (atomate).